scrapinghub/client/exceptions.py

```diff
@@ -69,6 +73,8 @@ def wrapped(*args, **kwargs):
                 raise ValueTooLarge(http_error=exc)
             elif 400 <= status_code < 500:
                 raise ScrapinghubAPIError(http_error=exc)
+            elif 500 <= status_code < 600:
+                raise ServerError(http_error=exc)
```

Comment:
@vshlapakov depending on whether it's possible to receive a non 200 response with some json body that should proceed to `self._decode_response` (not sure what's the return code for e.g. this error https://github.com/scrapinghub/python-scrapinghub/pull/55/files#diff-cd5ba6b088815a563525dde1900bbfecR150)